languages/go/exercises/concept/methods/.meta/design.md

```diff
@@ -0,0 +1,20 @@
+## Goal
+
+The goal of this exercise is to teach the student about `methods` in Go.
+
+## Learning objectives
+
+- Know what a method is.
+- Know how to write a method in Go.
+- Know that a method can be attached to any custom type in the same package (not just structs).
+- Know about a pointer vs value receiver.
```

Comment:
Would this be part of the `methods` concept, or is this a separate concept?

Reply:
It is part of how methods are attached to a type. If you have a value receiver, you can't change the values e.g. of a struct in the method.

This code will not update other copies of the struct. Just the current one.
```go
func (s someStruct) Update(str string) {
    s.someString = str
}
```

This however behaves similar to changing a property of a class in the C#:

```go
func (s *someStruct) Update(str string) {
    s.someString = str
}
```

Reply:
So... I'd say: Yes. is part of `methods`.

Reply:
Okay! Thanks for explaining